plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryAuthResolver.java

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.client;
+
+import com.google.inject.Inject;
+
+import org.apache.commons.codec.binary.Base64;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfigs;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for preparing docker auth data.
+ *
+ * @author Mykola Morhun
+ */
+public class DockerRegistryAuthResolver {
+
+    public static final String DEFAULT_REGISTRY = "";
+    public static final String DEFAULT_REGISTRY_VALUE = "docker.io";
+
+    private final InitialAuthConfig initialAuthConfig;
+
+    @Inject
+    public DockerRegistryAuthResolver(InitialAuthConfig initialAuthConfig) {
+        this.initialAuthConfig = initialAuthConfig;
+    }
+
+    /**
+     * Looks for auth header for specified registry and encode it in base64.
+     * First searches in the params and then in the initial auth config.
+     * If nothing found empty encoded json will be returned.
+     *
+     * @param registry
+     *         registry to which API call will be applied
+     * @param paramsAuthConfigs
+     *         auth data for current API call
+     * @return base64 encoded X-Registry-Auth header value
+     */
+    public String getXRegistryAuthHeaderValue(String registry, @Nullable AuthConfigs paramsAuthConfigs) {
+        Map<String, AuthConfig> authConfigsMap = null;
+        if (paramsAuthConfigs != null) {
+            authConfigsMap = paramsAuthConfigs.getConfigs();
+        }
+
+        AuthConfig authConfig = null;
+        if (authConfigsMap != null) {
+            for(Map.Entry<String, AuthConfig> entry : authConfigsMap.entrySet()) {
+                AuthConfig value = entry.getValue();
+                if (value.getServeraddress().contains(registry)) {
+                    authConfig = value;
+                    break;
+                }
+            }
+        }
+
+        if (authConfig == null) {
+            if (DEFAULT_REGISTRY.equals(registry)) {
```

Comment:
I think it can be good improvement to have a set with default registry names: docker.io, index.docker.io, http://indev.docker.io/v1/, "" so we will allow all of these as a default resitry identifier from auth conf provided as a parameter of this method